### PR TITLE
Police multiple attempts to set the one true Node instance

### PR DESCRIPTION
**Pull request by daviddrysdale**
Somewhat speculative, not sure that this is an ideal approach.

### CODE REVIEW

File: rust/oak/src/lib.rs [hunk 1]
```diff
@@ -107,22 +96,36 @@ thread_local! {
 /// }
 /// ```
 pub fn set_node<T: Node + 'static>() {
-    // TODO: Detect multiple invocations.
     NODE.with(|node| {
-        *node.borrow_mut() = Box::new(T::new());
+        match *node.borrow_mut() {
+            Some(ref mut _node) => {
```
Comment (tiziano88) [inline rust/oak/src/lib.rs]: Can this be `Some(_)` ?

File: rust/oak/src/lib.rs [hunk 2]
```diff
@@ -107,22 +96,36 @@ thread_local! {
...
+            let mut grpc_method_name = String::new();
+            grpc_method_channel
+                .read_to_string(&mut grpc_method_name)
+                .unwrap();
+            let mut grpc_channel = Channel::new(GRPC_CHANNEL_HANDLE);
+            node.invoke(&grpc_method_name, &mut grpc_channel);
+        }
+        None => {
+            let mut logging_channel = logging_channel();
+            writeln!(logging_channel, "gRPC call with no loaded Node").unwrap();
```
Comment (tiziano88) [inline rust/oak/src/lib.rs]: In other places `logging_channel()` was inlined, perhaps keep it consistent.

File: rust/oak/src/lib.rs (hunk 1 above)
Comment (daviddrysdale) [reply rust/oak/src/lib.rs]: Done.

File: rust/oak/src/lib.rs (hunk 2 above)
Comment (daviddrysdale) [reply rust/oak/src/lib.rs]: Done.

**Pull request merged**